client_update.go

```diff
@@ -86,8 +86,10 @@ func (u *UpdateClient) FetchUpdate(api ApiRequester, url string) (io.ReadCloser,
 	}
 
 	if r.ContentLength < 0 {
+		defer r.Body.Close()
```

Comment:
What about closing this couple of lines earlier - here: https://github.com/mendersoftware/mender/pull/95/files#diff-c1a9cef21663a7f28d8d2eb327d47bffR80